Fixes #912 where a file extension is required for some linters

cixtor opened the pull request:
I found that some linters like [SublimeLinter-phpcs](https://github.com/SublimeLinter/SublimeLinter-phpcs) stopped working after a recent update. After some debugging of the code I found that most recent versions of PHP CodeSniffer do not work when the file has a different extension than `.php` or has no extension at all.

Before SublimeLinter executes the command specified by the variable `cmd` it creates a temporary file, which on macOS is placed at `$TMPDIR` and then copies the content of the original file there, then it executes the command using `process.Popen`; the problem here is that the files are being created using `tempfile.NamedTemporaryFile` which by default creates file names with this format `/tmp[a-z_]{6}`.

A solution is to pass the `suffix=".ext"` parameter to the method call.

Fixes #912

FichteFoll commented:
I would rather fix `lint.Linter.get_tempfile_suffix` to not return an empty string when a view has a file name. That seems weird in the first place.

FichteFoll commented on lint/linter.py:
```diff
@@ -1298,18 +1298,22 @@ def run(self, cmd, code):
-            return suffix
-        else:
-            return ''
+        # default to plain text
```
I don't think we *need* an extension. Just leave it empty if none was specified. 
Alternatively, you could try inferring it from the view's file name, if that exists.

cixtor replied:
Is this the logic that you are suggesting for that method?

1. Extract the extension from `self.view.file_name()` and return if available
2. Infer the extension from the currently selected syntax and return if possible
3. Use `self.tempfile_suffix` in any other cases
4. Never return empty

Please confirm so I can commit these changes.

DaveStein commented:
@cixtor My files have a `.php` extension so not sure it relates to your fix

kaste commented on lint/util.py:
```diff
@@ -336,7 +336,7 @@ def tmpfile(cmd, code, filename, suffix='', output_stream=STREAM_STDOUT, env=Non
         filename = os.path.splitext(filename)[0] + suffix
 
-    file = tempfile.NamedTemporaryFile(delete=False)
+    file = tempfile.NamedTemporaryFile(delete=False, suffix=suffix)
```
Bc this is a regression from https://github.com/SublimeLinter/SublimeLinter/commit/a3db729367c0ceb423a3f617ab3fa8ab4b739409

(we just forgot about the suffix completely) This line alone should restore the old SublimeLinter behavior. So if you do just that and it works again. I could merge immediately bc I don't have to think through this whole thing. 

I don't remember we touched `get_tempfile_suffix` recently (git blame?). So I don't want to start a change here for what appears to be hotfixable. WDYT?

Of course the initial lines about `filename`in that function are no-ops. `filename` is not used anymore. Say, we had a bad day here.

braver replied:
Damn, who tests these things. And why does git `blame` me for these lines? 😒

The hotfix (and removal of those lines about "filename") seems like the way to go to me. Can someone confirm the fix for the php thing?

cixtor commented:
@DaveStein my PHP files also have a `.php` extension.

If you read again my description of issue #913 you will see that the problem is not the file itself but the temporary file that SublimeLinter is creating at `$TMPDIR` which is the file is that actually passed to `phpcs` for linting, because PHP CodeSniffer excepts the file to have a `.php` extension to work, it fails and returns no warnings or errors.

As @kaste said, this pull-request will _(probably)_ fix both GitHub issues as they are the same thing.

@kaste — @braver  I have updated the pull-request with the deletion of the unused `filename` variable and associated logic. And added the missing `suffix=suffix` to the `tempfile.NamedTemporaryFile()` method call. Let me know if there is anything else missing.

braver commented:
Yes, this should fix #913 and #912. I assume this means it fixed your #912, so I don't have to set up a phpcs test environment for myself. LGTM, so I'll merge and push a release so people get to enjoy php code sniffing again.
